sdk/environments/environment.go

```diff
@@ -165,6 +165,10 @@ func (e *ApiEndpoint) withResourceIdentifier(identifier string) *ApiEndpoint {
 	return e
 }
 
+func (e *ApiEndpoint) Available() bool {
+	return e != nil && (e.resourceIdentifier != nil || e.endpoint != nil)
```

Comment:
We'd need the Endpoint to know where to connect to, and the Resource Identifier to connect to it, so presumably should be:

```suggestion
	return e != nil && (e.resourceIdentifier != nil && e.endpoint != nil)
```

?

Reply:
Actually we can ignore the `resourceIdentifier` since we auto-fallback to use the `endpoint` e.g. in [scopes.go](https://github.com/hashicorp/go-azure-sdk/blob/b3ce3103af8e0d71698e7a16553bc7e06c796f9d/sdk/environments/scopes.go#L27-L40)